core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/ArrangedWriter.java

```diff
@@ -293,58 +299,74 @@ private boolean isStillReferenced(SubjectInContext key) {
 
 	private synchronized void queueStatement(Statement st) {
 		SubjectInContext key = new SubjectInContext(st);
-		Set<Statement> stmts = stmtBySubject.get(key);
+		Set<Statement> stmts = statementBySubject.get(key);
 		if (stmts == null && st.getSubject() instanceof BNode && !stack.contains(key)) {
 			blanks.add(st);
 		} else {
 			if (stmts == null) {
-				stmtBySubject.put(key, stmts = new TreeSet<>(comparator));
+				statementBySubject.put(key, stmts = new TreeSet<>(comparator));
 			}
 			stmts.add(st);
 		}
 		queueSize++;
 	}
 
 	private synchronized void flushStatements() throws RDFHandlerException {
-		if (!stmtBySubject.isEmpty() || !blanks.isEmpty()) {
+		if (!statementBySubject.isEmpty() || !blanks.isEmpty()) {
 			flushNamespaces();
-			Statement st;
 
 			// used to store all the statements
-			ArrayList<Statement> statements = new ArrayList<Statement>();
+			ArrayList<Statement> statements = new ArrayList<>();
 
 			// used to store blank nodes along with the number of times they are used as an object in a statement.
-			Map<BNode, Integer> bNodeOccurence = new HashMap<BNode, Integer>();
+			Map<BNode, Integer> bNodeOccurrences = new HashMap<>();
 
+			Statement st;
 			while ((st = nextStatement()) != null) {
 				statements.add(st);
+
 				Value obj = st.getObject();
 
-				// if the object in the statement is a blank node, we will update its number of occurence
+				// if the object in the statement is a blank node, we will update its number of occurrences
 				if (obj instanceof BNode) {
-					BNode bNode = (BNode) obj;
-					if (!bNodeOccurence.containsKey(bNode)) {
-						bNodeOccurence.put(bNode, 0);
+					bNodeOccurrences.compute(
+							(BNode) obj,
+							(key, i) -> i == null ? 1 : i + 1
+					);
+
+					if (bNodeOccurrences.get(obj) > 1) {
+						noneInlinedNodes.add(st.getSubject());
```

Comment:
Shouldn't you check if the subject is a blank node before adding it?

Reply:
I've tweaked this locally and it seems to solve the inlining issue where you had to adjust the test case, without compromising on your own new test case. I'll push a change.

Reply:
Not sure actually. But if it works better now than I'm fine with it.